### PR TITLE
require changes to workspace acl for all user group to be performed by admin

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/DSDEEPB-2061

1) prevent any changes to workspace ACLs wrt the all users group via the normal means
2) create admin api endpoints to inspect, grant and revoke read access to a workspace for the all users gorup
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceACL.scala [hunk 1]
```diff
@@ -10,20 +10,18 @@ case class WorkspaceACLUpdate(email: String, accessLevel: WorkspaceAccessLevel)
 
 object WorkspaceAccessLevels {
   sealed trait WorkspaceAccessLevel extends RawlsEnumeration[WorkspaceAccessLevel] with Ordered[WorkspaceAccessLevel] {
-    val all = Seq(UnknownUser, NoAccess, Read, Write, Owner)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceACL.scala:13]: move this to a more accessible (and sensible) spot

File: src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceACL.scala [hunk 2]
```diff
@@ -10,20 +10,18 @@ case class WorkspaceACLUpdate(email: String, accessLevel: WorkspaceAccessLevel)
...
-
     def compare(that: WorkspaceAccessLevel) = { all.indexOf(this).compare(all.indexOf(that)) }
 
     override def toString = WorkspaceAccessLevels.toString(this)
     override def withName(name: String) = WorkspaceAccessLevels.withName(name)
   }
 
-  case object UnknownUser extends WorkspaceAccessLevel
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/WorkspaceACL.scala:21]: UnknownUser does not make sense as a workspace access level and I don't see it used anywhere.

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 3]
```diff
@@ -326,37 +342,43 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
...
-          workspaceContext.workspace.accessLevels.foreach { case (level, groupRef) =>
-            withRawlsGroup(groupRef, txn) { group =>
-              //remove existing records for users and groups in the acl update list
-              val users = group.users.filter( userRef => !allTheRefs.contains(userRef) )
-              val groups = group.subGroups.filter( groupRef => !allTheRefs.contains(groupRef) )
-
-              //generate the list of new references
-              val newusers = groupsByLevel.getOrElse(level, Map.empty).keys.collect({ case Left(ru) => RawlsUser.toRef(ru) })
-              val newgroups = groupsByLevel.getOrElse(level, Map.empty).keys.collect({ case Right(rg) => RawlsGroup.toRef(rg) })
+          if (allTheRefs.contains(UserService.allUsersGroupRef)) {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:345]: this is where acl changes are prevented

the rest of the changes in this method are just indenting

**Comment by helgridly** [comment]
:+1:

**Pull request merged**